app/src/main/java/com/infomaniak/drive/ui/fileList/fileDetails/SelectCategoriesFragment.kt

```diff
@@ -22,134 +22,145 @@ import android.view.LayoutInflater
 import com.infomaniak.drive.ui.fileList.fileDetails.CategoriesAdapter.UICategory
+import com.infomaniak.drive.ui.fileList.fileDetails.SelectCategoriesFragment.UsageMode.*
 import com.infomaniak.drive.utils.*
 import com.infomaniak.drive.views.DebouncingTextWatcher
+import com.infomaniak.lib.core.models.ApiResponse
 import kotlinx.android.synthetic.main.fragment_select_categories.*
 import kotlinx.android.synthetic.main.item_search_view.*
+import java.util.*
 
 class SelectCategoriesFragment : Fragment() {
 
     private val selectCategoriesViewModel: SelectCategoriesViewModel by viewModels()
     private val navigationArgs: SelectCategoriesFragmentArgs by navArgs()
 
     private lateinit var categoriesAdapter: CategoriesAdapter
+    private lateinit var usageMode: UsageMode
     private lateinit var file: File
+    private lateinit var selectedCategories: List<Category>
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
         inflater.inflate(R.layout.fragment_select_categories, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        file = FileController.getFileById(navigationArgs.fileId) ?: run {
+        initializeData()
+        if (usageMode == NO_CATEGORIES) {
             findNavController().popBackStack()
             return
         }
 
-        DriveInfosController.getCategoryRights().let {
+        DriveInfosController.getCategoryRights()?.apply {
+            if (usageMode == SELECTED_CATEGORIES) {
+                canCreateCategory = false
+                canDeleteCategory = false
+                canEditCategory = false
+            }
+        }.let {
             setCategoriesAdapter(it?.canEditCategory == true, it?.canDeleteCategory == true)
             setAddCategoryButton(it?.canCreateCategory == true)
         }
 
         searchView.hint = getString(R.string.searchTitle)
 
-        setListeners()
+        configureToolbar()
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) { setBackNavResult() }
+        createCategoryRow.setOnClickListener { navigateToCreateCategory() }
+        configureSearchView()
         setBackActionHandlers()
     }
 
+    private fun initializeData() {
+        with(navigationArgs) {
+            val tempFile = FileController.getFileById(fileId)
+            usageMode = if (fileId == -1 || tempFile == null) {
+                if (categories == null) {
+                    NO_CATEGORIES
+                } else {
+                    selectedCategories = DriveInfosController.getCurrentDriveCategoriesFromIds(categories?.toTypedArray()!!)
+                    SELECTED_CATEGORIES
+                }
+            } else {
+                file = tempFile
+                FILE_CATEGORIES
+            }
+        }
+    }
```

Comment:
```suggestion
    private fun initializeData() = with(navigationArgs) {
        val tempFile = FileController.getFileById(fileId)
        usageMode = if (fileId == -1 || tempFile == null) {
            if (categories == null) {
                NO_CATEGORIES
            } else {
                selectedCategories = DriveInfosController.getCurrentDriveCategoriesFromIds(categories?.toTypedArray()!!)
                SELECTED_CATEGORIES
            }
        } else {
            file = tempFile
            FILE_CATEGORIES
        }
    }
```

Reply:
This part has been refactored in the meantime.